plugins/woocommerce/src/Blocks/Utils/BlocksSharedState.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\Utils;
+
+use InvalidArgumentException;
+
+/**
+ * Manages the registration of interactivity config and state that is commonly shared by WooCommerce blocks.
+ * Initialization only happens on the first call to initialize_shared_config.
+ * Intended to be used as a singleton.
+ */
+trait BlocksSharedState {
+
+	/**
+	 * The consent statement for using private APIs of this class.
+	 *
+	 * @var string
+	 */
+	private static $consent_statement = 'I acknowledge that using private APIs means my theme or plugin will inevitably break in the next version of WooCommerce';
+
+	/**
+	 * The namespace for the config.
+	 *
+	 * @var string
+	 */
+	private static $settings_namespace = 'woocommerce';
+
+	/**
+	 * Whether the core config has been registered.
+	 *
+	 * @var boolean
+	 */
+	private static $core_config_registered = false;
+
+	/**
+	 * Cart state.
+	 *
+	 * @var mixed
+	 */
+	private static $blocks_shared_cart_state;
+
+	/**
+	 * Check that the consent statement was passed.
+	 *
+	 * @param string $consent_statement - The consent statement string.
+	 * @return true
+	 * @throws \InvalidArgumentException - If the statement does not match the class consent statement string.
+	 */
+	private static function check_consent( $consent_statement ) {
+		if ( $consent_statement !== self::$consent_statement ) {
+			throw new \InvalidArgumentException( 'This method cannot be called without consenting the API may change.' );
```

Comment:
We already imported `InvalidArgumentException` above.
```suggestion
			throw new InvalidArgumentException( 'This method cannot be called without consenting the API may change.' );
```